js/components/dataset/resource/list.vue

```diff
@@ -155,13 +161,17 @@ export default {
     },
     events: {
         'uploader:progress': function(id, uploaded, total) {
-            const el = this.$el.getElementById(`progress-${id}`);
+            const el = document.getElementById(`progress-${id}`);
             el.style.width = `${Math.round(uploaded * 100 / total)}%`;
         },
-        'uploader:complete': function(id, response) {
-            const file = this.$uploader.getFile(id);
-            this.files.$remove(this.files.indexOf(file));
+        'uploader:complete': function(id, response, file) {
+            this.files.splice(this.files.indexOf(file), 1);
```

Comment:
What is the intent of this line? Loos like we are modifying the files list in place but not sure why.

The previously used `$remove` invocation did not work for any specific reason?

Reply:
Yes, we remove the file from the array, and for Vue.js to be able to detect the change, we need to use `splice()` (See https://v1.vuejs.org/guide/list.html#Array-Change-Detection)
The previous [`$remove()`](https://012.vuejs.org/guide/list.html#Augmented_Methods) was removed (:wink:) from Vue.js 1.0.

Reply:
My bad, the syntax is just not the same (and there is not a dedicated section about it): https://v1.vuejs.org/guide/list.html#Caveats
I wil change to `this.files.$remove(file)` and see if it's working :+1: